src/utils.ts

```diff
@@ -160,35 +147,27 @@ export async function getMessagesByUserIn(client: MatrixClient, sender: string,
         }
         return messages;
     }
-
-    // The recommended APIs for fetching events from a room is to use both rooms/initialSync then /messages.
-    // Unfortunately, this results in code that is rather hard to read, as these two APIs employ very different data structures.
-    // We prefer discarding the results from rooms/initialSync and reading only from /messages,
-    // even if it's a little slower, for the sake of code maintenance.
-    const timeline = response['messages']
-    if (timeline) {
-        // The end of the PaginationChunk has the most recent events from rooms/initialSync.
-        // This token is required be present in the PagintionChunk from rooms/initialSync.
-        let token = timeline['end']!;
-        // We check that we have the token because rooms/messages is not required to provide one
-        // and will not provide one when there is no more history to paginate.
-        while (token && processed < limit) {
-            const bfMessages = await backfill(token);
-            let lastToken = token;
-            token = bfMessages['end'];
-            if (lastToken === token) {
-                LogService.debug("utils", "Backfill returned same end token - returning early.");
-                return;
-            }
-            const events = filterEvents(bfMessages['chunk'] || []);
-            // If we are using a glob, there may be no relevant events in this chunk.
-            if (events.length > 0) {
-                await cb(events);
-            }
+    // We check that we have the token because rooms/messages is not required to provide one
+    // and will not provide one when there is no more history to paginate.
+    let token: string|null = null;
```

Comment:
We paginate backwards, so we never want to use this token again because it is far in the past next time we will want to be in the future. It might be an optimization to say "We should store the token to see if we get to this point next time and can finish", but that is problematic to implement and may not work at all since there's no guarantee for you to end up in the same spot in the stream next time.

Reply:
If it's difficult to optimize for, don't worry about it. At best, if you think that a comment would be useful for the next time someone ventures in this code, leave one.

Reply:
It doesn't seem right putting the comment here because you can't make this optimization to `getMessagesByUserIn` alone. This optimization would be specific to redactions because it is valid to want to paginate into history you have "seen before" in other situations.